src/Interfaces.jl

```diff
@@ -94,22 +94,57 @@ function methodparts(IT, x::Expr)
     return methodname, args
 end
 
-requiredmethod(IT, nm, args, shouldthrow) = :(Interfaces.implemented($nm, $args, mods) || ($shouldthrow && Interfaces.missingmethod($IT, $nm, $args, mods)))
+"""
+Construct the method signature from a function name and argument types.
+
+    methodsig(:foo, Tuple{Int64, Float64, Any}) == :(foo(::Int64, ::Float64, ::Any))
+"""
+methodsig(f, args) = Expr(:call, f, unconvertargs(args)...)
+
+function requiredmethod(IT, nm, args, shouldthrow)
+    :(Interfaces.implemented($nm, $args, mods) || ($shouldthrow && Interfaces.missingmethod(debuglvl, $IT, $nm, $args, mods)))
+end
 
-function requiredreturn(T, nm, args, shouldthrow, RT_sym, __RT__)
+function requiredreturn(IT, nm, args, shouldthrow, RT_sym, __RT__)
     return quote
-        check = $(requiredmethod(T, nm, args, shouldthrow))
```

Comment:
Hmmmm.....at first glance, it seems we should be keeping the `check = ` here?

Reply:
good spot! looks left over from some debugging :) thanks